tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,748 @@
+
+
+async def test_flow_auth_os_error(hass):
+    """Test we handle an OS error in the auth step."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.auth.side_effect = OSError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "auth"
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_reset_works(hass):
+    """Test we finish a config flow after a factory reset."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.auth.side_effect = blke.AuthenticationError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    with patch("broadlink.discover", return_value=[device.get_mock_api()]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"] == device.get_entry_data()
+
+
+async def test_flow_unlock_works(hass):
+    """Test we finish a config flow with an unlock request."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.is_locked = True
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "unlock"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"unlock": True},
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"] == device.get_entry_data()
+
+    assert mock_api.set_lock.call_args == call(False)
+    assert mock_api.set_lock.call_count == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)